### PR TITLE
ZIOS-11279: Go to a specific conversation when clicking on a link outside/inside the app

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

Added `.openConversation` action to load a conversation when opening a `wire://conversation/` link. It uses the same behavior of opening the conversation from a push notification. Based on https://github.com/wireapp/wire-ios/pull/3275.

### CODE REVIEW

**Comment by billypchan** [comment]
@nicorsm I tested the case "launch with the deep link" and it passed.

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 1]
```diff
@@ -576,9 +576,14 @@ extension AppRootViewController: SessionManagerURLHandlerDelegate {
...
-        case .openConversation(_, _):
-        ///TODO: open a conversation if id is valid
-            break
+        case .openConversation(_, let conversation):
+            if let conversation = conversation,
+               let userSession = ZMUserSession.shared() {
+                sessionManager?.showConversation(conversation, at: nil, in: userSession)
+                return true
+            } else {
+                return false
```
Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: @billypchan  we return `false` which if I remember correctly won't remove the pending url action. Won't we then potentially have a loop were we continuously try perform this action?

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 2]
```diff
@@ -576,9 +576,14 @@ extension AppRootViewController: SessionManagerURLHandlerDelegate {
 
     func sessionManagerShouldExecuteURLAction(_ action: URLAction, callback: @escaping (Bool) -> Void) -> Bool {
         switch action {
-        case .openConversation(_, _):
-        ///TODO: open a conversation if id is valid
-            break
+        case .openConversation(_, let conversation):
+            if let conversation = conversation,
+               let userSession = ZMUserSession.shared() {
+                sessionManager?.showConversation(conversation, at: nil, in: userSession)
```
Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: Why is it necessary to go through the session manager here and not call `zClientViewController.showConversation()` like we do for user profiles?

Comment (billypchan) [reply Wire-iOS/Sources/AppRootViewController.swift]: I think the reason is the `SessionManager` does not provide an interface without `userSession` to open a conversation. I will update the interface.

Comment (billypchan) [reply Wire-iOS/Sources/AppRootViewController.swift]: @typfel After refactoring in https://github.com/wireapp/wire-ios-sync-engine/pull/988, both open user profile/conservation action will be handled by SessionManager's methods.

**Pull request merged**